var/spack/repos/builtin/packages/racket/package.py

```diff
@@ -11,28 +11,57 @@ class Racket(Package):
 
     homepage = "https://www.racket-lang.org"
 
-    maintainers = ['arjunguha']
+    maintainers = ['arjunguha', 'elfprince13']
 
-    version('8.3.0', 'c4af1a10b957e5fa0daac2b5ad785cda79805f76d11482f550626fa68f07b949')
+    version('8.3', '3b963cd29ae119e1acc2c6dc4781bd9f25027979589caaae3fdfc021aac2324b')
 
     depends_on('libffi', type=('build', 'link', 'run'))
     depends_on('patchutils')
+    depends_on('libtool', type=('build'))
 
     phases = ['configure', 'build', 'install']
 
     def url_for_version(self, version):
-        url = "http://mirror.racket-lang.org/installers/{0}/racket-src.tgz"
-        return url.format(version.up_to(2))
+        return "https://mirror.racket-lang.org/installers/{0}/racket-minimal-{0}-src-builtpkgs.tgz".format(version)
+
+    variant('cs', default=True, description='Build Racket CS (new ChezScheme VM)')
+    variant('bc', default=False, description='Build Racket BC (old MZScheme VM)')
+    variant('shared', default=True, description="Enable shared")
+    variant('jit', default=True, description="Just-in-Time Compilation")
+
+    parallel = False
+    extendable = True
+
+    def toggle(self, spec, variant):
+        toggle_text = ("enable" if spec.variants[variant].value else "disable")
+        return "--{0}-{1}".format(toggle_text, variant)
 
     def configure(self, spec, prefix):
         with working_dir('src'):
-            configure = Executable('./configure')
-            configure("--prefix", prefix)
+            configure = Executable("./configure")
+            configure_args = [self.toggle(spec, 'cs'),
+                              self.toggle(spec, 'bc'),
+                              self.toggle(spec, 'jit')]
+            toggle_shared = self.toggle(spec,'shared')
```

Comment:
Our style checks require a space after each function arg.

Reply:
Oops, thought I had fixed that, but should be addressed in 2d68635